common/src/autogluon/common/utils/log_utils.py

```diff
@@ -193,3 +193,32 @@ def reset_logger_for_remote_call(verbosity: int):
     # Limiting the verbosity of the BaggedEnsembleModel logger to 10 to avoid
     # duplicated messages about fitting.
     set_logger_verbosity(verbosity=min(verbosity, 1), logger=bem_logger)
+
+
+def warn_if_mlflow_autologging_is_enabled(logger: Optional[logging.Logger] = None):
+    """Log a warning if MLflow autologging is enabled.
+
+    MLflow autologging monkey-patches the sklearn metrics, which leads to a PicklingError when AutoGluon models
+    that have a metric as an instance attribute are saved to disk.
+
+    Related issues:
+        - https://github.com/mlflow/mlflow/issues/6268
+        - https://github.com/autogluon/autogluon/issues/4914
+    """
+    if logger is None:
+        logger = _logger_ag
+    try:
+        import mlflow
+
+        try:
+            if not mlflow.utils.autologging_utils.autologging_is_disabled("sklearn"):
+                logger.warning(
+                    "⚠️ Warning: MLflow autologging is enabled. This will likely break AutoGluon model training. "
+                    "Please disable autologging by executing `import mlflow; mlflow.autolog(disable=True)` before importing AutoGluon."
+                )
+        except Exception:
+            # gracefully handle cases where autologging_is_disabled is not available
+            pass
+    except Exception:
```

Comment:
This could, in theory, be narrowed down to an `ImportWarning`, but I don't want to accidentally break the entire AG training process if `import mlflow` fails with some other spectacular exception.